roadmap/README.md

```diff
@@ -0,0 +1,148 @@
+# Tekton Mission and Roadmap
+
+This doc describes Tekton's mission and the 2020 roadmap primarily
+for Tekton Pipelines but some sneak peeks at other projects as well.
+
+- [Mission and Vision](#mission-and-vision)
+- [Roadmap](#roadmap)
+  - [Beta and GA releases](#beta-and-ga)
+  - ["Feature Complete" Tekton](#feature-complete-tekton)
+    - [Task Interfaces and PipelineResources](#task-interfaces-and-pipelineresources)
+  - [SCM support](#scm-support)
+  - [Results](#results)
+  - [Release and dogfooding](#release-and-dogfooding)
+- Tekton Projects
+  - [Triggers](./triggers.md)
+  - [Catalog](./catalog.md)
+
+## Mission and Vision
+
+Tekton's mission:
+
+  Be the industry-standard, cloud-native CI/CD platform components and ecosystem.
+
+The vision for this is:
+
+* Tekton API conformance across as many CI/CD platforms as possible
+* A rich catalog of high quality, reusable `Tasks` which work with Tekton conformant systems
+
+What this vision looks like differs across different users:
+
+* **Engineers building CI/CD systems**: These users will be motivated to use Tekton
+  and integrate it into the CI/CD systems they are using because building on top
+  of Tekton means they don't have to re-invent the wheel and out of the box they get
+  scalable, serverless cloud native execution
+* **Engineers who need CI/CD**: (aka all software engineers!) These users will benefit
+  from the rich high quality catalog of reusable components:
+
+  * Quickly build and interact with sophisticated `Pipelines`
+  * Be able to port `Pipelines` to any Tekton conformant system
+  * Be able to use multiple Tekton conformant systems instead of being locked into one
+    or being forced to build glue between multiple completely different systems
+  * Use an ecosystem of tools that know how to interact with Tekton components, e.g.
+    IDE integrations, linting, CLIs, security and policy systems
+
+## Roadmap
+
+### Beta and GA
+
+In 2019 we got to the point where we had several projects built on top of Tekton
+that were craving additional API stability, so we started a push toward having
+a beta release of Tekton. We have
+[created a plan](https://docs.google.com/document/d/1H8I2Rk4kLdQaR4mV0A71Qbk-1FxXFrmvisEAjLKT6H0/edit)
+which defines what Beta means for Tekton Pipelines, and
+[through our beta working group](https://github.com/tektoncd/community/blob/master/working-groups.md#beta-release)
+we are working towared a beta release in early 2020.
+
+The initial beta release
+[does not include all Tekton Pipeline components](https://docs.google.com/document/d/1H8I2Rk4kLdQaR4mV0A71Qbk-1FxXFrmvisEAjLKT6H0/edit#heading=h.t0sc4hdrr5yq). After our initial
+beta release, we will then work toward:
+
+1. Beta for the remaining components
+1. Beta for [Tekton Triggers](#triggers)
+1. GA for both Tekton Pipelines and Tekton Triggers
```

Comment:
We should be *less* specific on the beta on the tekton level. I would see something more like

> We target a beta release for the core component earlier this year, leading to a full GA (1.0) release of a defined set of components

^^ we still need to discuss what we want to do for this, but my feeling is, as the project grows, we may want to have the main components released together and *maybe* share the same version number (easier to manage).

In a `Tekton` GA, I wanna see pipeline, triggers, cli and most likely the dashboard ; but in addition to that, I feel we should also target a versionned catalog that has a well defined process (validation, support, …)

Reply:
Maybe we can get all the projects to release together, I'd like to weigh the pros and cons, I'm not 100% convinced - perhaps what I can put here can be more general, about making some decisions around this?

I do think it makes sense to be explicit about a beta release of Triggers tho for example.